src/targets/gpu/include/migraphx/gpu/context.hpp

```diff
@@ -265,12 +267,43 @@ struct context
         this->current_device = std::make_shared<hip_device>(0, n_streams);
     }
 
+    void wait_for(any_ptr queue)
+    {
+        hipStream_t sPtr          = queue.get<hipStream_t>();
+        hip_device::stream ctx_q  = get_stream();
+
+        auto status = hipEventRecord(start_event.get(), sPtr);
+        if(status != hipSuccess)
+        {
+            std::stringstream ss;
+            ss << "wait_for: failed to record " << status;
+            MIGRAPHX_THROW(ss.str());
```

Comment:
Just do `MIGRAPHX_THROW("failed to record: " + hip_error(status))`. `hip_error` converts the hip error to a string. Also, `MIGRAPHX_THROW` will add the function name to the message automatically.

Reply:
Will do. Didn't realize that we convert hip_error. That's really handy